internal/kafka/internal/handlers/validation.go

```diff
@@ -145,21 +208,50 @@ func ValidateKafkaUserFacingUpdateFields(ctx context.Context, authService author
 	}
 }
 
-func ValidateKafkaClaims(ctx context.Context, kafkaRequestPayload *public.KafkaRequestPayload, kafkaRequest *dbapi.KafkaRequest) handlers.Validate {
-	return func() *errors.ServiceError {
-		kafkaRequest = presenters.ConvertKafkaRequest(*kafkaRequestPayload, kafkaRequest)
-		claims, err := auth.GetClaimsFromContext(ctx)
-		if err != nil {
-			return errors.Unauthenticated("user not authenticated")
+func getClaims(ctx context.Context) (jwt.MapClaims, *errors.ServiceError) {
+	claims, err := auth.GetClaimsFromContext(ctx)
+	if err != nil {
+		return nil, errors.Unauthenticated("user not authenticated")
+	}
+	return claims, nil
+}
+
+type ValidateKafkaClaimsOptions func(claims *jwt.MapClaims) *errors.ServiceError
+
+func ValidateUsername() ValidateKafkaClaimsOptions {
+	return func(claims *jwt.MapClaims) *errors.ServiceError {
+		username := auth.GetUsernameFromClaims(*claims)
+		if username == "" {
+			return errors.New(errors.ErrorBadRequest, "can't find 'user' attribute in claims")
 		}
-		(*kafkaRequest).Owner = auth.GetUsernameFromClaims(claims)
-		(*kafkaRequest).OrganisationId = auth.GetOrgIdFromClaims(claims)
-		(*kafkaRequest).OwnerAccountId = auth.GetAccountIdFromClaims(claims)
+		return nil
+	}
+}
 
+func ValidateOrganisationId() ValidateKafkaClaimsOptions {
+	return func(claims *jwt.MapClaims) *errors.ServiceError {
+		orgId := auth.GetOrgIdFromClaims(*claims)
+		if orgId == "" {
+			return errors.New(errors.ErrorBadRequest, "can't find 'organisation id' attribute in claims")
```

Comment:
the `organisation id` attribute can be confusing:
The claim key in use are defined in:
- https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/4e3302d57225475aad5e54e286041946fe8a4c8b/pkg/auth/context.go#L26
- or the alternative one in https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/4e3302d57225475aad5e54e286041946fe8a4c8b/pkg/auth/context.go#L35

and these can be configured as well i.e they are dynamic so I think we should use the right claims key or reword the message a bit to avoid mentioning wrong claim key

Reply:
The message is already generic and is not mentioning the claim key. However, I believe it should be coherent with the same error about the `username`, so I'm going to change it here to include correct key names

Reply:
Thanks @ziccardi